lib/rubocop/cop/rails/file_path.rb

```diff
@@ -48,6 +48,7 @@ class FilePath < Base
         def on_dstr(node)
           return unless rails_root_nodes?(node)
           return unless node.children.last.str_type?
+          return if node.children.last.source.start_with?(':')
```

Comment:
This causes `Metrics/AbcSize` warning. I don't have a good idea to avoid the cop...

```
lib/rubocop/cop/rails/file_path.rb:48:9: C: Metrics/AbcSize: Assignment Branch Condition size for on_dstr is too high. [<0, 17, 5> 17.72/17]
        def on_dstr(node) ...
        ^^^^^^^^^^^^^^^^^
```

Reply:
How about the following?

```diff
diff --git a/lib/rubocop/cop/rails/file_path.rb b/lib/rubocop/cop/rails/file_path.rb
index 540c8d314..9a9e3a09e 100644
--- a/lib/rubocop/cop/rails/file_path.rb
+++ b/lib/rubocop/cop/rails/file_path.rb
@@ -48,9 +48,11 @@ module RuboCop
         def on_dstr(node)
           return unless rails_root_nodes?(node)
           return unless node.children.last.str_type?
-          return if node.children.last.source.start_with?(':')
-          return unless node.children.last.source.start_with?('.') ||
-                        node.children.last.source.include?(File::SEPARATOR)
+
+          last_child_source = node.children.last.source
+          return unless last_child_source.start_with?('.') ||
+                        last_child_source.include?(File::SEPARATOR)
+          return if last_child_source.start_with?(':')

           register_offense(node)
         end
```

Reply:
Thanks!